### PR TITLE
feature: made `Storage` to be a super trait

### LINKED ISSUE #24: Use super traits to reduce boiler plate in the EC store.
Currently, there is a whole bunch of:

```rust
where
    S: Storage + Send + Sync + 'static
```

which can just be simplified.

### PR DESCRIPTION
**Pull request by druzn3k**
`Storage` now implements explicitly also the traits `Send + Sync + 'static`, as they are needed everywhere `Storage` is needed. Thanks to @ChillFish8 for the guidance.

### CODE REVIEW

**Comment by druzn3k** [comment]
Don't mind the name of the super trait, it's meant to be temporary till discussed with you.

**Comment by ChillFish8** [comment]
I see what you mean, although I have a different set of false positives.

I think i've fixed it by simplifying the trait to just require the super traits on the storage trait itself. We can do this since you need to meet these requirements anyway.

**Comment by druzn3k** [comment]
Thank you, I will take a look at your commit right away.

**Comment by druzn3k** [comment]
Ok @ChillFish8, I've seen the modification. It was the first approach that I wanted to do but discarded because I thought that was done like this for a reason and I didn't want to break existing codes. I will edit the PR title and text to match the current implementation.

**Comment by ChillFish8** [comment]
>  It was the first approach that I wanted to do but discarded because I thought that was done like this for a reason and I didn't want to break existing codes.

Ah no, it was just left over from incrementally building it and the required constraints. Let me know when this is good to go :)

**Comment by druzn3k** [comment]
For me it's good to go, if the title and descriptions fits the required standard.

File: datacake-eventual-consistency/src/test_utils.rs
Comment (ChillFish8) [inline datacake-eventual-consistency/src/test_utils.rs]: Just noticed this, but we can't do this as A) we dont have a justification for why it's safe and B) this should be auto-implemented and can be removed.

**Review by ChillFish8 — CHANGES_REQUESTED** [review]

File: datacake-eventual-consistency/src/test_utils.rs
Comment (druzn3k) [reply datacake-eventual-consistency/src/test_utils.rs]: Ops, my bad. It's an experiment that got committed! :(

**Review by ChillFish8 — APPROVED** [review]

**Pull request merged**
